Add 2023-08-16 agenda and minutes (#556)
solid/specification: meetings/2023-08-16.md

* SC: Can anyone interested add to this issue please, and then it might inform more specificity in the Solid protocol, and how much variability we expect here.
* TBL: The protocol must give guarantees. The Solid protocol gives guarantees about the state of the resource on the Web. The ETag protocol is the best way of doing so. If it breaks, than you can lose data. We have tests, so that data will still be safe.

### Chat Client-Client spec new work item
URL: https://github.com/solid/specification/issues/553

* TBL: Does anyone have an objection to the proposal?
* SC: We are evaluating the proposal.
* WT: My point is the one I'm making in topic below. IMO the CG and WG are not places to define domain specific interop specs.
* SC: Why not?
* WT: Please see discussion for next topic.
* TBL: Client-Client specs have always been part of the design plans. We also have code in SolidOS. If they are not included in CG, we can still do it as part of Solid Project.
* SC: It is worth reviewing the objections. My understanding is that it is within CG scope to have it as work item.
* VB: +1

> 

Reviewer: @hzbarcea, I believe at the end of this point, you said something about not being sure whether I was properly understood, since the next issue indeed shows my belief in client-client interop within Solid. That sentence was however not recorded in the minutes, making it seem like I am indeed against any such endeavour. Do you think the following addition captures what you said correctly? If not, please suggest a better wording. Thanks!

```suggestion
* Hadrian: I think we should first look into the next topic (issue #554), because there Wouter seems to support client-client interoperability. I'm not sure how this matches what he is saying here.
```

Reviewer: @csarven: This should suffice, but I leave it up to @hzbarcea whether or not to include it, since it are his words. 

Alternatively, I would propose that we include the following at the end of the topic: _"Let it be noted that, because of lack of time, Wouter did not have the opportunity to respond to this."_

Author: Excellent. Yes, I vaguely remember this near the end of the meeting and yes, @hzbarcea , it'd be great if you can confirm or adjust as you see fit.

* eP: What's the format you have it currently in? Can you add it to an issue (#48)?
* WT: I have in markdown so I can just do it.
* SC: It seems like useful exercise / information so we can review it. I think it would be better not to have a central location where all the specs are referring to. This would be unusual. Solid Team was working on solid glossary. The spec usually defines it own term like Resource Server. Later one can use SKOS to match concepts across documents. I don't think we should have a central document with all definitions.
* WT: Maybe we can have defined a couple of main concepts. Instead having each spec redefining it.
* SC: Some concepts are unclear if spec is defining it or reusing it.
* SC: A discussion topic would be a good place to post that information, and tag the editors of the specs in question.
* eP: For me it would be interesting to see which are actually product classes. Can you highlight those, composing multiple product classes in single implementation seems common. I think we still do not have a clear view on how that should be done.
* SC: But this is a subtopic. Not sure if it necessarily resolves it.

> * WT: This was more concrete issue, I think current definition of Server is very broad. I don't think is wrong but for conformance class should use something like Resource Server.

Reviewer: ```suggestion
* WT: This was more concrete issue, I think current definition of Server is very broad. I don't think is wrong but for conformance class should use something like Resource Server. OAuth has a good definition of that term.
```

### Add access management lifecycle to scope
URL: https://github.com/solid/solid-wg-charter/pull/46

* SC: We agreed that the CG will push these decision and reach consensus on the WG charter proposal and assign it to PA Champin, so that he has acknowledgement about what's there. He is on vacation now until end of week, so I was going to propose this, even if it is no major change.
* eP: I recall that PA was participating and he sees it as just clarifying the scope. I thought we assigned to him because SC was creating most pulls and would have to approve his own, but this PR was created by A Coburn.
* SC: ...
* SC: Any objection to merging?
* SC: No objection.

ACTION: SC will merge and link.

### Rename 'Server' to something more specific — 'Resource Server' or 'Storage Server' or ...
URL: https://github.com/solid/specification/issues/548

* SC: First two are reasonable.
* WT: On a slightly more general note: I started a document comparing the shared definitions of all current reports. Can we have that as an organic definition document to refer to? Will create a PR (where?).
* SC: PR against what document?
* WT: Either in the protocol document or separate document.

> * SC: Are you mapping the concepts, e.g., agent - agent
> * WT: I am mapping - looked all the TRs - if two are using "resource server" for example, what the definitions are and if they are same or not. As a proposal to use the same definition.

Reviewer 2: ```suggestion
* SC: Are you mapping the concepts, e.g., agent — agent
* WT: I am mapping — looked all the TRs — if two are using "resource server", for example, what the definitions are, and if they are same or not. As a proposal to use the same definition.
```

* eP: What's the format you have it currently in? Can you add it to an issue (#48)?
* WT: I have in markdown so I can just do it.
* SC: It seems like useful exercise / information so we can review it. I think it would be better not to have a central location where all the specs are referring to. This would be unusual. Solid Team was working on solid glossary. The spec usually defines it own term like Resource Server. Later one can use SKOS to match concepts across documents. I don't think we should have a central document with all definitions.
* WT: Maybe we can have defined a couple of main concepts. Instead having each spec redefining it.
* SC: Some concepts are unclear if spec is defining it or reusing it.
* SC: A discussion topic would be a good place to post that information, and tag the editors of the specs in question.

> * eP: For me it would be interesting to see which are actually product classes. Can you highlight those, composing multiple product classes in single implementation seems common. I think we still do not have a clear view on how that should be done.

Reviewer 2: ```suggestion
* eP: For me it would be interesting to see which are actually product classes. Can you highlight those? Composing multiple product classes in a single implementation seems common. I think we still do not have a clear view on how that should be done.
```

* SC: But this is a subtopic. Not sure if it necessarily resolves it.
* WT: This was more concrete issue, I think current definition of Server is very broad. I don't think is wrong but for conformance class should use something like Resource Server.
* SC: ... realises the more general issue: https://github.com/solid/specification/issues/480
* SC: Can we move this till after Wouter's proposal?

> * eP: I think Resource Server is a great example to start from. It's already used in a few reports, and we also use Solid Storage a lot. Solid Notifications is not directly depend on the Solid spec, but we assume that solid storage will implement notifications resource server.

Reviewer 2: ```suggestion
* eP: I think Resource Server is a great example to start from. It's already used in a few reports, and we also use Solid Storage a lot. Solid Notifications is not directly dependent on the Solid spec, but we assume that solid storage will implement notifications resource server.
```

ACTION: WT posts in discussion

### Return ETag headers on PUT requests
URL: https://github.com/CommunitySolidServer/CommunitySolidServer/issues/632

* SC: Let's have additional eyes/minds on this. Cases in which the ETag header can(not) be used in `PUT` and `PATCH` responses. Potentially introduce a requirement or add advisory in Solid Protocol.
* SC: General issue is that CSS might want more input on how it should approach. I raise this here so that people can have a look, both at CSS and Solid Protocol, to possibly refine the spec.
* WT: What is the added value for the Solid ecosystem, on top of the affordances already provided by HTTP?

> * SC: It's about whether the interpretation of the spec is correct? Is it within what the Solid protocol allows? Or is it required?

Reviewer 2: ```suggestion
* SC: It's about whether the interpretation of the spec is correct. Is it within what the Solid protocol allows? Or is it required?
```

* WT: Given that current solid spec doesn't require anything about it, it is implementation choice, what would be the advantage of mandating it by the spec?
* SC: I thought ETag was required.
* TBL: For the logic to work we need ETag to work. It provides conditional writes. (explains Etag...)
* SC: Solid says it is a MUST. Clients can expect ETag.

> * TBL: Clients need to rely on it. Is this issue on whether it is also required on put (unless the ETag RFC already requires it).

Reviewer 2: ```suggestion
* TBL: Clients need to rely on it. Is this issue on whether it is also required on PUT (unless the ETag RFC already requires it)?
```

* SC: Solid protocol does not distinguish between the method; it relies on the RFC. The question is what the behavior is on PUT and PATCH.

> * AC: I want to make sure that any decision we make is consistent with current HTTP spec and how current browsers work. Etags work on representations. If a server returns a 204 we do not want the browser to cache that. We also don't want it to conflict with HTTP: it is really focused on representations.
> * WT: Wanted to join AC on that about representations. What I meant earlier what is not in the current protocol, for GET, ETag is clearly there. Same could go for PUT, based on representation, it can return an ETag.

Reviewer 2: ```suggestion
* AC: I want to make sure that any decision we make is consistent with current HTTP spec and how current browsers work. Etags work on representations. If a server returns a 204, we do not want the browser to cache that. We also don't want it to conflict with HTTP; it is really focused on representations.
* WT: Wanted to join AC on that about representations. What I meant earlier, what is not in the current protocol; for GET, ETag is clearly there. Same could go for PUT; based on representation, it can return an ETag.
```

* eP: I want to ask about PATCH, since it used a different content type. Which representation would be returned?

> * SC: It depends on that of the representation type of the response; whether the payload has a non-empty content or content-location header, e.g., 201 or 200. It seems like PUT or PATCH responses could thus fall under this case. Is that interpretation correct? What are implementations and browsers currently doing? 

Reviewer 2: ```suggestion
* SC: It depends on the representation type of the response; whether the payload has a non-empty content or content-location header, e.g., 201 or 200. It seems like PUT or PATCH responses could thus fall under this case. Is that interpretation correct? What are implementations and browsers currently doing? 
```

* SC: POST is, I think, not the case, but it is a bit special in Solid because of container treatment.

> * TBL: The Etag has to do with representation, not with the body of the response but with the state of the resource. With a PUT the state is the body of the request. In a PATCH has changed the state, the client needs to be able to update its Etag. The logic is always: whenever a resource state changes, the Etag changes.

Reviewer 2: ```suggestion
* TBL: The Etag has to do with representation, not with the body of the response, but with the state of the resource. With a PUT, the state is the body of the request. In a PATCH, the state has changed, the client needs to be able to update its Etag. The logic is always: whenever a resource state changes, the Etag changes.
```

* AC: Agreed. I just want to make sure what we do does not break how browsers currently behave.
* TBL: Browsers typically don't use a PUT.

> * AC: I mean, what if we use JavaScript, that's going to use the browsers cache.

Reviewer 2: ```suggestion
* AC: I mean, what if we use JavaScript, that's going to use the browser's cache.
```

* TBL: It would be useful to have an overview of what we expect browsers to do.

> * SC: Sounds like a moving target. Like CORS. There is so much to do. We have a panel around QA work; we really need people. For us to show interop we need to put the work in, either in the CG or in the WG. That includes the browser behavior: if we have any problems with how the web platform works, we can actually raise those, e.g., to the WICG

Reviewer 2: ```suggestion
* SC: Sounds like a moving target. Like CORS. There is so much to do. We have a panel around QA work; we really need people. For us to show interop, we need to put the work in, either in the CG or in the WG. That includes the browser behavior: if we have any problems with how the web platform works, we can actually raise those, e.g., to the WICG.
```

* RG: I want you to read [8.8.3 of RFC9110](https://httpwg.org/specs/rfc9110#field.etag)
  > The "ETag" field in a response provides the current entity tag for the selected representation, as determined at the conclusion of handling the request. An entity tag is an opaque validator for differentiating between multiple representations of the same resource, regardless of whether those multiple representations are due to resource state changes over time, content negotiation resulting in multiple representations being valid at the same time, or both.
* SC: I think there is indeed a nuance with the strong and weak Etags. What we currently have in the spec is what we came up with, but it is not set in stone. If you have some time, please read through them to not repeat anything. If there is new material, that might warrant a change.
* SC: Can anyone interested add to this issue please, and then it might inform more specificity in the Solid protocol, and how much variability we expect here.

> * TBL: The protocol it must give guarantees. The Solid protocol gives guarantees about the state of the resource on the Web. The Etag protocol is the best way of doing so. If it breaks, than you can lose data. We have tests, so that data will still be safe.

Reviewer 2: ```suggestion
* TBL: The protocol must give guarantees. The Solid protocol gives guarantees about the state of the resource on the Web. The Etag protocol is the best way of doing so. If it breaks, than you can lose data. We have tests, so that data will still be safe.
```

### Chat Client-Client spec new work item
URL: https://github.com/solid/specification/issues/553

* TBL: Does anyone have an objection to the proposal?
* SC: We are evaluating the proposal.

> * WT: My point is the one I'm making in topic below. IMO the CG noro WG are places to define domain specific interop specs.

Reviewer 2: ```suggestion
* WT: My point is the one I'm making in topic below. IMO the CG and WG are not places to define domain-specific interop specs.
```

* SC: Why not?
* WT: Please see discussion for next topic.

> * TBL: Client-Client specs has always been part of the design plans. We also have code in SolidOS. If they are not included in CG we can still do it as part of Solid Project.

Reviewer 2: ```suggestion
* TBL: Client-Client specs have always been part of the design plans. We also have code in SolidOS. If they are not included in CG, we can still do it as part of Solid Project.
```

> * SC: It is worth to review the objections. My understanding is that it is within CG scope to have it as work item.

Reviewer 2: ```suggestion
* SC: It is worth reviewing the objections. My understanding is that it is within CG scope to have it as work item.
```
